internal/factsengine/gatherers/verifypassword.go

```diff
@@ -9,45 +9,52 @@ import (
 	crypt "github.com/tredoe/osutil/user/crypt"
 	sha512crypt "github.com/tredoe/osutil/user/crypt/sha512_crypt"
 	"github.com/trento-project/agent/pkg/factsengine/entities"
+	"github.com/trento-project/agent/pkg/utils"
 )
 
 const (
 	VerifyPasswordGathererName = "verify_password"
+	checkableUsernames         = "hacluster"
```

Comment:
This could be a map of username => wellKnownPassword 
```go
var wellKnownUserPassowrdPairs = map[string]string{
	"hacluster": "linux",
}
```
It would allow to easily add more pairs.

Reply:
This would force us to duplicate common passwords for every user we want to check; rather than a map I'd have a separate array of passwords, wdyt?

Reply:
If I get it right, the duplication that is concerning you is the following
```go
var wellKnownUserPassowrdPairs = map[string]string{
	"hacluster":      "linux",
	"anotheruser":    "linux",
	"yetanotheruser": "linux",
}
```
I could live with that for the time being, yet I see your point.
Feel free to come with changes that address review's feedbacks and the hardcoded password duplication concern.

If it becomes too much of a hassle, consider deferring to a later stage the complexity of linking a list/map of usernames to a list/map of passwords.

Reply:
I proposed something slightly different in the code, take a look and let me know what you think :+1:

Reply:
I'd find just more natural working with lists/maps for this usecase, rather than strings that we need to split or look into.
Let's get some feedbacks from others as well.

Reply:
@nelsonkopliku it depends on the usage. if we agree that we check the same set of passwords for all the users, then a map is not the right solution.